### DISCUSSION TITLE
Did this way been banned by the lastest DSM7.2.1 69057 update5?

### DISCUSSION THREAD

**Discussion by fanzixuan** [007revad/Synology_HDD_db#216] [Q&A]
I recently bought a DS923+, tried couple of times with this method, however everything seems ok until after rebooting, i still cannot add my ssd as storage pool, the compatibility warning is still there, i'm pretty sure i've done it with exactly steps, and i also notice for synology's release note(69057 update) shows something like"security upgrade", is it related?
Thanks a lot! I'm close to be mad...

**Comment by fanzixuan** [comment]
add:
/ ssd has been recognized with their right model

Correct
/ it should be "DSM 7.2.1 69057 update3"

**Comment by fanzixuan** [comment]
![image](https://github.com/007revad/Synology_HDD_db/assets/8508935/118274b9-c2c5-4fa0-8f83-b49bb5d09932)
name after GLOWAY... is correct, i don't know what "HX/MAXIO" means, seems irrelavant..

**Comment by fanzixuan** [comment]
![image](https://github.com/007revad/Synology_HDD_db/assets/8508935/0160bc7f-4e47-415e-a445-615b5c297480)
script running success screenshot

**Comment by 007revad** [comment]
> i don't know what "HX/MAXIO" means, seems irrelavant

HX/MAXIO is the company that your NVMe drive's vendor id are registered to. There are 2 companies registered to that vendor id: HX and MAXIO.

If DSM doesn't have that vendor id in it's list storage manager will show "Unknown" as the brand, and the "Unknown firmware version" warning which prevents creating a volume on the NVMe drive.

The script works in DSM 7.2.1 69057 update 3

**Comment by arbitarytwist** [comment]
Having a same issue with FIKWOT drives, which show as HX/MAXIO.

Were you able to resolve this ? Does not show unknown but cannot create storage pool

**Reply by fanzixuan** [reply]
yes bro, it's fixable, i did it in the end, however i quit NASing for a while, cannot remember the details,,,

**Comment by 007revad** [comment]
It seems there are a lot NVMe brands that are all made by the same company: HX, MAXIO, Lexar, FIKWOT, GLOWAY etc. So they all use the same `0x05dc` vendor id code.

You can change it to FIKWOT with:
```
sudo synosetkeyvalue /usr/syno/etc.defaults/pci_vendor_ids.conf 0x05dc FIKWOT
sudo synosetkeyvalue /usr/syno/etc/pci_vendor_ids.conf 0x05dc FIKWOT
```

**Comment by arbitarytwist** [comment]
These are the VID & SSVID i get. 
vid     : 0x1e4b
ssvid   : 0x1e4b

Do i still use 0x05dc

**Reply by 007revad** [reply]
Use 0x1e4b
```
sudo synosetkeyvalue /usr/syno/etc.defaults/pci_vendor_ids.conf 0x1e4b FIKWOT
sudo synosetkeyvalue /usr/syno/etc/pci_vendor_ids.conf 0x1e4b FIKWOT
```

**Comment by arbitarytwist** [comment]
I tried that with the unknown vendor being HX/MAXIO or FIKWOT. The unknown goes away, but i cannot use the drive as i have the warning in storage manager "This drive has not been tested or validated for M.2 SSD storage pools".

EDIT: I also tried the --force option but no changes.

**Reply by 007revad** [reply]
Can you reply with the output from syno_hdd_db.

**Reply by arbitarytwist** [reply]
@gNAS:/volume1/Data/HDD_db-3.5.96$ sudo -s ./syno_hdd_db.sh -nr
Synology_HDD_db v3.5.96
DS723+ DSM 7.2.1-69057-5
StorageManager 1.0.0-0017

ds723+_host_v7 version 8041

Using options: -nr
Running from: /volume1/Data/HDD_db-3.5.96/syno_hdd_db.sh

HDD/SSD models found: 1
Blue SA510 2.5 1000GB,52020100,1000 GB

M.2 drive models found: 1
FIKWOT FN501 Pro 512GB,SN12221,512 GB

No M.2 PCIe cards found

No Expansion Units found

Blue SA510 2.5 1000GB already exists in ds723+_host_v7.db
FIKWOT FN501 Pro 512GB already exists in ds723+_host_v7.db

Support disk compatibility already enabled.

Support memory compatibility already disabled.

Max memory is set to 32 GB.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already disabled.

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.

**Reply by arbitarytwist** [reply]
I also reset / re-installed the DSM for a fresh start but same result.

**Comment by arbitarytwist** [comment]
Hello. Wondering if you have any other thoughts on things to try.

**Reply by 007revad** [reply]
Can you run the script with the -s option so I can see what the database entry looks like.
```
./syno_hdd_db.sh -nrs
```
Does "Storage Manager > HDD/SSD" show: FIKWOT FIKWOT FN501 Pro 512GB

Or: FIKWOT FN501 Pro 512GB

**Reply by arbitarytwist** [reply]
See below.

Yes the storage manager does show it as FIKWOT FIKWOT Pro 512GB (SSD).

Interestingly after the command, "./syno_hdd_db.sh -nrs" i was able to create the volume. Didn't even reboot it. 

```
w@gNAS:/volume1/Data/HDD_db-3.5.96$ sudo ./syno_hdd_db.sh -nrs
Password:
Synology_HDD_db v3.5.96
DS723+ DSM 7.2.1-69057-5
StorageManager 1.0.0-0017

ds723+_host_v7 version 8041

Using options: -nrs
Running from: /volume1/Data/HDD_db-3.5.96/syno_hdd_db.sh

HDD/SSD models found: 1
Blue SA510 2.5 1000GB,52020100,1000 GB

M.2 drive models found: 1
FIKWOT FN501 Pro 512GB,SN12221,512 GB

No M.2 PCIe cards found

No Expansion Units found

Blue SA510 2.5 1000GB already exists in ds723+_host_v7.db
FIKWOT FN501 Pro 512GB already exists in ds723+_host_v7.db

Support disk compatibility already enabled.

Support memory compatibility already disabled.

Max memory is set to 32 GB.

NVMe support already enabled.

M.2 volume support already enabled.

Drive db auto updates already disabled.

Blue SA510 2.5 1000GB:
{
  "52020100": {
    "size_gb": 1000,
    "compatibility_interval": [
      {
        "compatibility": "support",
        "not_yet_rolling_status": "support",
        "fw_dsm_update_status_notify": false,
        "barebone_installable": true,
        "barebone_installable_v2": "auto",
        "smart_test_ignore": false,
        "smart_attr_ignore": false
      }
    ]
  },
  "default": {
    "size_gb": 1000,
    "compatibility_interval": [
      {
        "compatibility": "support",
        "not_yet_rolling_status": "support",
        "fw_dsm_update_status_notify": false,
        "barebone_installable": true,
        "barebone_installable_v2": "auto",
        "smart_test_ignore": false,
        "smart_attr_ignore": false
      }
    ]
  }
}

FIKWOT FN501 Pro 512GB:
{
  "SN12221": {
    "size_gb": 512,
    "compatibility_interval": [
      {
        "compatibility": "support",
        "not_yet_rolling_status": "support",
        "fw_dsm_update_status_notify": false,
        "barebone_installable": true,
        "barebone_installable_v2": "auto",
        "smart_test_ignore": false,
        "smart_attr_ignore": false
      }
    ]
  },
  "default": {
    "size_gb": 512,
    "compatibility_interval": [
      {
        "compatibility": "support",
        "not_yet_rolling_status": "support",
        "fw_dsm_update_status_notify": false,
        "barebone_installable": true,
        "barebone_installable_v2": "auto",
        "smart_test_ignore": false,
        "smart_attr_ignore": false
      }
    ]
  }
}

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.
```

**Reply by 007revad** [reply]
> Interestingly after the command, "./syno_hdd_db.sh -nrs" i was able to create the volume. Didn't even reboot it

You just reminded me that about a year ago some people said they needed to run the script 2 or 3 times, when they first used it. I never did work out what the reason was.

Some of those people scheduled the script to run at shutdown and at boot.

**Comment by arbitarytwist** [comment]
Something weird happened.

I changed both the primary drives and installed DSM. The nvme although shows unknown was already a storage volume, whereas the drives didn't have a storage volume.

**Reply by 007revad** [reply]
When you reinstalled DSM it would mount any existing storage pools and volumes.

**Reply by arbitarytwist** [reply]
But if I removed both the drives, where is this information saved, so it can recreate it.

**Reply by 007revad** [reply]
When DSM creates a storage pool on a drive, including NVMe drives, it includes partition and file system information near the beginning of the drive.
